aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -341,8 +341,10 @@ Tensor& atan2_out(Tensor& result, const Tensor& self, const Tensor& other) {
 }
 
 Tensor atan2(const Tensor& self, const Tensor& other) {
-  Tensor result = at::empty({0}, self.options());
-  return native::atan2_out(result, self, other);
+  Tensor result;
+  auto iter = TensorIterator::binary_op(result, self, other);
```

Comment:
This is cool and completely correct. Would you mind updating it, though? We now have `TensorIterator::binary_float_op`, which applies to `atan2`. That makes it so integer inputs to `atan2` are always promoted to float values (just like `div`). 

Would you make `atan2` a `binary_float_op` and test that integer inputs are accepted and return the expected float values? This would require expanding your test in test_type_promotion and the list of types we test `atan2` to with here:

https://github.com/pytorch/pytorch/blob/d2b4534d4dfb3bf973db078c9b994a127f6fe79d/test/test_torch.py#L19881

Reply:
Sure. Thanks!